### DISCUSSION TITLE
Welcome to WebTorrent Discussions!

### DISCUSSION THREAD

**Discussion by feross** [webtorrent/webtorrent#2067] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by geolffreym** [comment]
@feross thanks for open this channel for community. 
We were working for a while on the development of our web application and we faced the interoperability barrier between bittorrent and webtorrent, I leave here a small post about the solution that we gave to this problem and an alternative for those who have a similar use case: https://dev.to/geolffreym/watchit-web-torrent-streaming-proposal-511d

**Reply by hello-smile6** [reply]
Nice!

**Comment by VictorV96** [comment]
Hello. Your example at https://webtorrent.io/ does not work for me on all networks. About a year ago there were feasts, and now the same webseed.
![image](https://user-images.githubusercontent.com/23615212/118402077-d6759a80-b681-11eb-953a-7b858a663f50.png)

At the moment I want to implement a full-fledged video hosting using webtorrent. At the moment I ran into a problem with nat. Are there any webtorrent modules to expand the network? Ie so that webtorrent-hybrid clients are proxies for finding peers for clients on webtorrent.js (webrtc).

**Comment by ps1ph3r** [comment]
@feross was wondering about some things about webtorrent. In traditional torrenting during initial seeding you have to port forward to accept any incoming connections (correct me if I am wrong as I started torrenting a few months back) maybe that was because of tcp. But in webtorrent I think this problem is solved as it uses webrtc. I don't know much. Please throw some light on this. Thanks a lot for webtorrent.

**Reply by DiegoRBaquero** [reply]
WebRTC solves this for WebRTC peers. For TCP/UDP peers there's this issue https://github.com/webtorrent/webtorrent/issues/195

**Comment by abdemirza** [comment]
How Can I use it with react ?
Please help me regarding this.

**Reply by DiegoRBaquero** [reply]
Webtorrent is pure JS, can be integrated in any JS (web) app. Outside of the scope to do an implementation ourselves, sorry.

**Reply by Bruce-Hopkins** [reply]
Is it possible to add React integration in the future?

**Reply by DiegoRBaquero** [reply]
What type of integration are you looking for? WebTorrent being pure JS can be integrated with any frontend framework.

Maybe see https://github.com/mikeal/webtorrent-element for more ideas.

**Reply by Bruce-Hopkins** [reply]
Oh I'm sorry I misunderstood your answer 😄

Could the repo benefit from a examples created with React? Or is the implementation too obvious?

**Reply by DiegoRBaquero** [reply]
The implementation varies a lot depending on what you are looking to do with it. General torrent app? Render/Stream a file? Single file sharing? Too many. See https://github.com/webtorrent/webtorrent/blob/master/docs/faq.md#who-is-using-webtorrent-today

**Comment by Adamanthus** [comment]
Is this project abandoned?

**Reply by DiegoRBaquero** [reply]
Nope, certainly not abandoned. But not under active development

**Comment by strijdhtie-tech** [comment]
good day, the current version 0.24 for mac is no longer supported by apple on  M chips. Can you recompile it for Apple Sillicon?  Thank you.
